mephisto/core/task_launcher.py

```diff
@@ -45,6 +54,10 @@ def __init__(
         self.assignments: List[Assignment] = []
         self.units: List[Unit] = []
         self.provider_type = task_run.get_provider().PROVIDER_TYPE
+        self.max_num_concurrent_units = max_num_concurrent_units
+        self.num_running_units = 0
```

Comment:
We don't need this value, as we can always check `len(self.launched_units)` and builtins already do this bookkeeping to keep the runtime of `len` constant and low.